### PR TITLE
🌱  Refactor Makefile and add proto compile support

### PR DESCRIPTION
**Pull request by azeemshaikh38**
* **Please check if the PR fulfills these requirements**
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] PR title follows the guidelines defined in  https://github.com/ossf/scorecard/blob/main/CONTRIBUTING.md#pr-process

* **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)
Add new features to Makefile.

* **What is the current behavior?** (You can also link to an open issue here)
 - Always install all dependencies.
 - No separate `make install` command.
 - Does not support compiling proto files.
 - Does not run Docker build during `make all`

* **What is the new behavior (if this is a feature change)?**
- Only installs dependencies, if an existing binary is not found.
- Adds a new `make install` command for all required dependencies.
- Adds support for compiling proto fiels.
- Runs Docker build during `make all`

* **Does this PR introduce a breaking change?** (What changes might users need to make in their application due to this PR?)

* **Other information**:

### CODE REVIEW

**Review by naveensrinivasan — APPROVED** [review]
Thanks for cleaning it up! 

There are conflicts

**Comment by azeemshaikh38** [comment]
> Thanks for cleaning it up!
> 
> There are conflicts

Done. Also took care of https://github.com/ossf/scorecard/pull/455#pullrequestreview-660277491

File: Makefile [hunk 1]
```diff
@@ -21,61 +26,125 @@ help:  ## Display this help
...
+install: | $(GINKGO) $(GOLANGGCI_LINT) $(PROTOC_GEN_GO) $(PROTOC)
+
+$(GINKGO):
+	go get -u github.com/onsi/ginkgo/ginkgo@v1.16.2
+
+$(GOLANGGCI_LINT):
+	go install github.com/golangci/golangci-lint/cmd/golangci-lint@v1.40.0
+
+$(PROTOC_GEN_GO):
+	go install google.golang.org/protobuf/cmd/protoc-gen-go
```
Comment (inferno-chromium) [inline Makefile:43]: Can you please pin the dependency version for security reasons, similar to others.

**Review by inferno-chromium — COMMENTED** [review]
Amazing, thanks for this cleanup!
this comment can be addressed in a future PR.

File: Makefile (hunk 1 above)
Comment (naveensrinivasan) [reply Makefile:43]: Here is the issue to track this https://github.com/ossf/scorecard/issues/459

**Pull request merged**